### DISCUSSION TITLE
Allow deleting history items from issues/PRs

### DISCUSSION THREAD

**Discussion by xnousnow** [community/community#47520] [Projects and Issues]
### Select Topic Area

Product Feedback

### TL;DR: We should be able to delete commit histories in GitHub issues.
In GitHub, we generally do make issues and commit to resolving them. Let's say we have this issue:
```
| ߦ Stranger added a commit that referenced this issue on Feb 10, 2023 |
| | feat: fix bygs (#9999)                                             |
```
Sometimes we rebase, undo, or amend **pushed commit**. above actions change the hash, even if we do not modify the contents of the commit. Let's say we amended to fix the typo, and the following happens:
```
| ߦ Stranger added a commit that referenced this issue on Feb 10, 2023 |
| | feat: fix bygs (#9999)                                     cc604b4 |
| ߦ Stranger added a commit that referenced this issue on Feb 10, 2023 |
| | feat: fix bugs (#9999)                                     e450ed3 |
```
You can see that the original commit with the typo still exists in history. **We should be able to delete history items like these in GitHub issues.** The example is just a fix for a typo, but think that You pushed the API key to your repo, and the commit is referenced to the issue. You can contact the support team, but we can't do that for every typo/rebased/squashed commit.

This issue is continued from [isaacs/github #270](https://github.com/isaacs/github/issues/270).
